### PR TITLE
[Analysis] Enumeration entries

### PR DESCRIPTION
**Pull request by serras**
This PR refactors and uses parts of the primary constructor check to check "enumeration entries."

### CODE REVIEW

File: plugins/liquid/refined-types-plugin/src/main/kotlin/arrow/meta/plugins/liquid/phases/analysis/solver/check/Expressions.kt [hunk 1]
```diff
@@ -696,6 +697,13 @@ private fun SolverState.checkConstantExpression(
   NoReturn
 }
 
+private fun String.toBigIntegerSeveralAttempts(): BigInteger? =
```
Comment (raulraja) [inline plugins/liquid/refined-types-plugin/src/main/kotlin/arrow/meta/plugins/liquid/phases/analysis/solver/check/Expressions.kt:700]: LOL what's this xD?

**Review by raulraja — APPROVED** [review]
Looks great @serras! exciting to see new declarations getting support.

File: plugins/liquid/refined-types-plugin/src/main/kotlin/arrow/meta/plugins/liquid/phases/analysis/solver/check/Expressions.kt (hunk 1 above)
Comment (serras) [reply plugins/liquid/refined-types-plugin/src/main/kotlin/arrow/meta/plugins/liquid/phases/analysis/solver/check/Expressions.kt]: Me after figuring out that after 30 minutes debugging it was only a matter of parsing an integer correctly xD

I've changed it to `asIntegerLiteral` and made it follow Kotlin's grammar.

Comment (serras) [reply plugins/liquid/refined-types-plugin/src/main/kotlin/arrow/meta/plugins/liquid/phases/analysis/solver/check/Expressions.kt]: But maybe it should be replaced with https://github.com/JetBrains/kotlin/blob/master/compiler/psi/src/org/jetbrains/kotlin/parsing/ParseUtils.kt altogether

**Pull request merged**